mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobDefinition.java

```diff
@@ -246,6 +252,26 @@ public Optional<ClusterID> getResourceCluster() {
             .map(l -> ClusterID.of(l.getValue()));
     }
 
+    /**
+     * Returns the map of assignment attributes deduced from labels. The attributes are extracted
+     * from labels matching the "SCHEDULING_CONSTRAINT_LABEL_REGEX" pattern, and only the capturing
+     * group content (i.e., the key that comes after "_mantis.schedulingConstraint.") is saved.
+     *
+     * The return value would be used as constraint during worker scheduling.
+     */
+    public Map<String, String> getAssignmentAttributes() {
+        return labels.entrySet().stream()
+            .map(label -> {
+                final Matcher matcher = SCHEDULING_CONSTRAINT_LABEL_REGEX.matcher(label.getKey());
+                return matcher.find() ? Pair.of(matcher.group(1), label.getValue().getValue()) : null;
+            })
+            .filter(Objects::nonNull)
+            .collect(Collectors.toMap(
+                Pair::getLeft,
+                Pair::getRight
+            ));
+    }
+
```

Comment:
nit: we can probably do this computation in the constructor given that this is a value class.